utils/src/main/java/io/neow3j/contract/NeoURI.java

```diff
@@ -0,0 +1,184 @@
+package io.neow3j.contract;
+
+import io.neow3j.utils.AddressUtils;
+import io.neow3j.utils.Strings;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class NeoURI {
+
+    private String uri;
+
+    protected NeoURI(Builder builder) {
+        this.uri = builder.uri;
+    }
+
+    public String getUri() {
```

Comment:
can you also return a `java.net.URI`?

Then, we would have:

`public URI getUri()`

`public URI getUriAsString()`

Reply:
Added both methods
(I replaced the URI string with a URI object that is generated in the builder.)